[![cloudevel](img/cloudevel.png)](https://cloudevel.com)

# Compute firewalls.

##  Firewalls.

https://cloud.google.com/vpc/docs/firewalls

In [12]:
gcloud compute networks get-effective-firewalls default --format=json

{[?2004l
  "firewalls": [
    {
      "allowed": [
        {
          "IPProtocol": "tcp",
          "ports": [
            "80"
          ]
        }
      ],
      "creationTimestamp": "2021-10-17T12:10:01.806-07:00",
      "description": "",
      "direction": "INGRESS",
      "disabled": false,
      "id": "4157466274877787110",
      "kind": "compute#firewall",
      "logConfig": {
        "enable": false
      },
      "name": "default-allow-http",
      "network": "https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/networks/default",
      "priority": 1000,
      "selfLink": "https://www.googleapis.com/compute/v1/projects/tercera-generacion/global/firewalls/default-allow-http",
      "sourceRanges": [
        "0.0.0.0/0"
      ],
      "targetTags": [
        "http-server"
      ]
    },
    {
      "allowed": [
        {
          "IPProtocol": "tcp",
          "ports": [
            "443"
          ]
        }
      ],
      "creationTimestamp": "2021-10

: 1

In [18]:
gcloud compute firewall-rules list

NAME                    NETWORK  DIRECTION  PRIORITY  ALLOW                         DENY  DISABLED
default-allow-http      default  INGRESS    1000      tcp:80                              False
default-allow-https     default  INGRESS    1000      tcp:443                             False
default-allow-icmp      default  INGRESS    65534     icmp                                False
default-allow-internal  default  INGRESS    65534     tcp:0-65535,udp:0-65535,icmp        False
default-allow-rdp       default  INGRESS    65534     tcp:3389                            False
default-allow-ssh       default  INGRESS    65534     tcp:22                              False

To show all fields of the firewall, please show in JSON format: --format=json
To show all fields in table format, please see the examples in --help.



: 1

### Reglas.

* ```Direction```: ```ingress```o ```egress```.
* ```Priority```: 0 a 65535
* ```Action```: ```allow```o ```deny```
* ```Target```: instancias.
* ```Source/destination``` 
* ```Protocol``` y ```port```: ```TCP```, ```UDP```o ```ICMP```.
* ```Enforment status```: ```enabled``` o ```disabled```.

2 reglas por defecto con prioridad 65535:
* Permite la salida a 0.0.0.0/0
* Niega tráfico de entrada a 0.0.0.0/0

#### Reglas al crear una VPC.

    Incoming traffic from any VM instance on the same network
    Incoming TCP traffic on port 22, allowing SSH
    Incoming TCP traffic on port 3389, allowing Microsoft Remote Desktop Protocol (RDP)
    Incoming Internet Control Message Protocol (ICMP) from any source on the network

```
gcloud compute firewall-rules
```

--action
--allow
--description
--destination-ranges
--direction
--network
--priority
--source-ranges
--source-service-accounts
--source-tags
--target-service-accounts
--target-tags

```
gcloud compute firewall-rules create ace-exam-fwr2 –-network ace-exam-vpc1 –-allow tcp:20000-25000
```

<p style="text-align: center"><a rel="license" href="http://creativecommons.org/licenses/by/4.0/"><img alt="Licencia Creative Commons" style="border-width:0" src="https://i.creativecommons.org/l/by/4.0/80x15.png" /></a><br />Esta obra está bajo una <a rel="license" href="http://creativecommons.org/licenses/by/4.0/">Licencia Creative Commons Atribución 4.0 Internacional</a>.</p>
<p style="text-align: center">&copy; José Luis Chiquete Valdivieso. 2021.</p>